src/bundle/Controller/TrashController.php

```diff
@@ -99,15 +104,19 @@ public function performAccessCheck(): void
      */
     public function listAction(Request $request): Response
     {
-        $page = $request->query->get('page') ?? 1;
-        $trashItemsList = [];
+        $searchForm = $this->formFactory->searchTrash();
+        $searchForm->handleRequest($request);
 
-        $query = new Query([
-            'sortClauses' => [new Query\SortClause\Location\Priority(Query::SORT_ASC)],
-        ]);
+        $page = $request->query->get('trash_search')['page'] ?? 1;
```

Comment:
I don't know from what this cames from, and if it an issue in this case, but when I moved search to new bundle it came out that all those queries are `strings`, so you may get some type issues somewhere.